src/OrchardCore.Modules/OrchardCore.Search.Lucene/Services/LuceneIndexManager.cs

```diff
@@ -95,6 +95,21 @@ public async Task DeleteDocumentsAsync(string indexName, IEnumerable<string> con
                 }
             });
         }
+        public async Task DeleteDocumentsByVersionIdAsync(string indexName, IEnumerable<string> ContentItemVersionIds)
```

Comment:
This is unnecessary as we always only store one version of a content item (contentItemid) in the indexes (Lucene, Elasticsearch). And also the contentItemId is the only field that we know that will never be null.

Reply:
removed